contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,202 @@
+    {}
+
+    /// @notice add an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the new address to add as a rateLimitedAddress
+    /// @param _rateLimitPerSecond the rate limit per second for this rateLimitedAddress
+    /// @param _bufferCap  the buffer cap for this rateLimitedAddress
+    function addAddress(address rateLimitedAddress, uint112 _rateLimitPerSecond, uint144 _bufferCap) external virtual override onlyGovernorOrAdmin {
+        require(_bufferCap <= bufferCap, "MultiRateLimited: new buffercap too high");
+
+        RateLimitData memory rateLimitData = RateLimitData({
+            lastBufferUsedTime: _blockTimestamp(),
+            bufferCap: _bufferCap,
+            rateLimitPerSecond: _rateLimitPerSecond,
+            bufferStored: _bufferCap
+        });
+
+        rateLimitPerAddress[rateLimitedAddress] = rateLimitData;
+
+        emit IndividualRateLimitPerSecondUpdate(rateLimitedAddress, 0, _rateLimitPerSecond);
+    }
+
+    /// @notice add an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the address whose buffer and rate limit per second will be set
+    /// @param _rateLimitPerSecond the new rate limit per second for this rateLimitedAddress
+    /// @param _bufferCap  the new buffer cap for this rateLimitedAddress
+    function updateAddress(address rateLimitedAddress, uint112 _rateLimitPerSecond, uint144 _bufferCap) external virtual override onlyGovernorOrAdmin {
+        require(_bufferCap <= bufferCap, "MultiRateLimited: buffercap too high");
+
+        uint112 oldRateLimitPerSecond = _rateLimitPerSecond;
+
+        RateLimitData storage rateLimitData = rateLimitPerAddress[rateLimitedAddress];
+
+        rateLimitData.lastBufferUsedTime = _blockTimestamp();
+        rateLimitData.bufferCap = _bufferCap;
+        rateLimitData.rateLimitPerSecond = _rateLimitPerSecond;
+        rateLimitData.bufferStored = _bufferCap;
+
+        emit IndividualRateLimitPerSecondUpdate(rateLimitedAddress, oldRateLimitPerSecond, _rateLimitPerSecond);
+    }
+
+    /// @notice remove an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the address to remove from the whitelist of addresses
+    function removeAddress(address rateLimitedAddress) external virtual override isGovernorOrGuardianOrAdmin {
+        uint256 oldRateLimitPerSecond = rateLimitPerAddress[rateLimitedAddress].rateLimitPerSecond;
+
+        delete rateLimitPerAddress[rateLimitedAddress];
+
+        emit IndividualRateLimitPerSecondUpdate(rateLimitedAddress, oldRateLimitPerSecond, 0);
+    }
+
+    /// @notice set the rate limit per second
+    /// @param rateLimitedAddress the address whose buffer will be set
+    /// @param newRateLimitPerSecond the new rate limit per second for this rateLimitedAddress
+    function setIndividualRateLimitPerSecond(address rateLimitedAddress, uint112 newRateLimitPerSecond) external virtual override onlyGovernorOrAdmin {
```

Comment:
would prefer syntax of setRateLimit(address, uint112) but not too attached to the idea

Reply:
In the RateLimitedContract, there is a method `setRateLimitPerSecond`, so I would like to keep this naming convention

Reply:
that sounds good, i'm mainly referring to the word individual

so perhaps setRateLimitPerSecond still? it'll be an overloaded function which seems fine

Reply:
looks like you did this, cool!